app/client/components/delivery/records/deliveryRecordsTable.tsx

```diff
@@ -0,0 +1,251 @@
+import { DeliveryRecordMessage } from "./deliveryRecordMessage";
+import { RecordAddress } from "./deliveryRecordsAddress";
+import { DeliveryProblemMap } from "./deliveryRecordsApi";
+import { PaginationNav } from "./deliveryRecordsPaginationNav";
+import { RecordStatus } from "./deliveryRecordStatus";
+
+interface RecordsTableProps {
+  data: DeliveryRecordsDetail[];
+  deliveryProblemMap: DeliveryProblemMap;
+  resultsPerPage: number;
+}
+
+export const RecordsTable = (props: RecordsTableProps) => {
+  const totalPages = Math.ceil(props.data.length / props.resultsPerPage);
+  const [currentPage, setCurrentPage] = useState(0);
+  const scrollToTop = () => {
+    window.scrollTo(0, 0);
+  };
+  const isOdd = (n: number) => Math.abs(n % 2) === 1;
+  const tbodyCSS = css`
+    td {
+      width: 100%;
+      display: block;
+      padding: ${space[2]}px ${space[2]}px ${space[5]}px;
+      vertical-align: top;
+      ${minWidth.tablet} {
+        width: auto;
+        display: table-cell;
+        text-align: left;
+        padding: ${space[2]}px 10px;
+      }
+    }
+    td[data-title] {
+      text-align: right;
+      ${minWidth.tablet} {
+        text-align: left;
+      }
+    }
+    td[data-title]:before {
+      content: attr(data-title);
+      font-weight: bold;
+      float: left;
+      ${minWidth.tablet} {
+        display: none;
+      }
+    }
+    td[data-title-block]:before {
+      content: attr(data-title-block);
+      font-weight: bold;
+      display: block;
+      ${minWidth.tablet} {
+        display: none;
+      }
+    }
+  `;
+
+  const trCSS = (rowNum: number, hasAdditionalMessages: boolean) => css`
+    display: flex;
+    flex-direction: column;
+    ${minWidth.tablet} {
+      display: table-row;
+      background: ${isOdd(rowNum) ? palette.neutral[97] : "none"};
+    }
+    td {
+      border-bottom: 1px solid ${palette.neutral[86]};
+      ${minWidth.tablet} {
+        border-bottom: ${hasAdditionalMessages
+          ? "none"
+          : `1px solid ${palette.neutral[86]}`};
+      }
+    }
+  `;
+
+  return !props.data.length ? (
+    <p>There aren't any delivery records to show you yet</p>
+  ) : (
+    <>
+      <table
+        css={css`
+          width: 100%;
+          ${textSans.medium()};
+          border-collapse: collapse;
+        `}
+      >
+        <thead
+          css={css`
+            display: none;
+            ${minWidth.tablet} {
+              display: table-header-group;
+            }
+            th {
+              text-align: left;
+              ${minWidth.tablet} {
+                padding: 0 10px ${space[2]}px;
+              }
+            }
+          `}
+        >
+          <tr>
+            <th>Date</th>
+            <th>Status</th>
+            <th>Delivery postcode</th>
+            <th>Delivery instructions</th>
+          </tr>
+        </thead>
+        <tbody css={tbodyCSS}>
+          {props.data
+            .filter(
+              (element, index) =>
```

Comment:
could we pull this function into a well-named `const` so it's easier to understand